File: backend/app/utils/ip_hash.py
import hashlib
import hmac
from datetime import datetime, timezone


def hash_ip(ip_address: str | None, secret: str) -> str | None:
    """Return a daily rotating HMAC-SHA256 digest instead of storing a raw IP."""
    if not ip_address:
        return None

    utc_date = datetime.now(timezone.utc).date().isoformat()
    message = f"{utc_date}:{ip_address}".encode("utf-8")
    return hmac.new(secret.encode("utf-8"), message, hashlib.sha256).hexdigest()

File: backend-express_1/backend_node/utils/ipHash.js
// Equivalent of app/utils/ip_hash.py
const crypto = require('crypto');

/** Return a daily rotating HMAC-SHA256 digest instead of storing a raw IP. */
function hashIp(ipAddress, secret) {
  if (!ipAddress) return null;
  const utcDate = new Date().toISOString().slice(0, 10); // YYYY-MM-DD, UTC
  const message = `${utcDate}:${ipAddress}`;
  return crypto.createHmac('sha256', secret).update(message).digest('hex');
}

module.exports = { hashIp };
